atoti-query-analyser/src/library/inputProcessors/v1tov2.js

```diff
@@ -308,131 +363,111 @@ const findFilter = (filters, needle) => {
-      r.timingInfo.startTime = [deps ? -deps.length : 0];
-    }
-  });
+  Object
+    .values(v1Structure.retrievals)
+    .sort((lhs, rhs) => lhs.sourceId.localeCompare(rhs.sourceId))
+    .forEach(retrieval => {
+      const { kind, retrievalId } = parseSourceId(retrieval);
+
+      let arrayToInsert;
+      let mappedRetrieval;
+
+      switch (kind) {
+        case "Retrieval":
+          arrayToInsert = result.aggregateRetrievals;
+          mappedRetrieval = mapAggregateRetrieval(retrieval, filters);
+          break;
+        case "ExternalRetrieval":
+          arrayToInsert = result.externalRetrievals;
+          mappedRetrieval = mapExternalRetrieval(retrieval, filters);
+          break;
+        default:
+          throw new Error(`Unexpected retrieval kind: ${kind}`);
+      }
 
-  const queue = retrievals
-    .filter(r => r.type.includes("NoOp") || r.timingInfo.startTime[0] === 0)
-    .map(r => r.retrId);
-  let safeCount = 0;
-  while (queue.length > 0) {
-    if (safeCount > 5000) {
-      console.log("oops. Too much");
-      return;
-    }
-    safeCount += 1;
-
-    const id = queue.shift();
-    dependents.get(id).forEach(did => {
-      const parent = retrMap.get(did);
-      if (did !== -1 && ++parent.timingInfo.startTime[0] === 0) {
-        parent.timingInfo.startTime[0] = computeFakeStart(
-          did,
-          retrMap,
-          dependencies
-        );
-        queue.push(did);
+      if (retrievalId !== arrayToInsert.length) {
+        throw new Error(`Cannot insert ${retrieval.sourceId} because ${kind}#${retrievalId - 1} not found`);
       }
+
+      arrayToInsert.push(mappedRetrieval);
     });
-  }
 
-  const incomplete = retrievals
-    .filter(r => r.timingInfo.startTime && r.timingInfo.startTime[0] < 0)
-    .map(r => r.retrId);
-  if (incomplete.length > 0) {
-    throw new Error(`Remaining retrievals without timings: ${incomplete}`);
-  }
+  return result;
 };
 
-const createSummary = retrievals => {
-  const measures = _(retrievals)
+const createSummary = (aggregateRetrievals, externalRetrievals) => {
+  // TODO Add externalRetrievals info
+  const measures = _(aggregateRetrievals)
     .flatMap(r => r.measures)
     .uniq()
     .value();
 
-  const retrievalsCountByType = _.countBy(retrievals, r => r.type);
-  const partitioningCountByType = _.countBy(retrievals, r => r.partitioning);
+  const retrievalsCountByType = _.countBy(aggregateRetrievals, r => r.type);
+  retrievalsCountByType["ExternalDatabaseRetrieval"] = _.size(externalRetrievals);
+  const partitioningCountByType = _.countBy(aggregateRetrievals, r => r.partitioning);
 
   return {
     measures,
-    totalRetrievals: _.size(retrievals),
+    totalRetrievals: _.size(aggregateRetrievals) + _.size(externalRetrievals),
     retrievalsCountByType,
-    partitioningCountByType
+    partitioningCountByType,
+    resultSizeByPartitioning: {}, // ???
+    partialProviders: [], // ???
+    totalExternalResultSize: 0, // ???
```

Comment:
Do you want to discuss these, as I see `??` in your code?

Reply:
No, I was wondering if it is correct to fill these fields with zeros.